test/unit_tests/braket/experimental/autoqasm/test_converters.py

```diff
@@ -59,3 +59,32 @@ def fn() -> None:
 a = 1;
 e = a;"""
     assert qasm == expected_qasm
+
+
+def test_break_for_loop():
+    @aq.function
+    def main():
+        for i in aq.range(3):
+            aq.gates.h(i)
+            break
+
+    with pytest.raises(aq.errors.UnsupportedFeature):
+        main()
+
+
+def test_break_while_loop():
+    @aq.function
+    def uses_while():
+        while aq.gates.measure(0):
+            aq.gates.x(0)
+
+    uses_while()
+
+    @aq.function
+    def uses_while_w_break():
+        while aq.gates.measure(0):
+            aq.gates.x(0)
+            break
+
+    with pytest.raises(aq.errors.UnsupportedFeature):
+        uses_while_w_break()
```

Comment:
I am not sure that `break` statements in `while` loops are actually unsupported. I think it might work. (autograph converts the `break` condition in `for` loops to an `extra_test` function that we need to handle, but there is no such thing in `while` loops - I think it transpiles the `break` condition into part of the loop condition.)

However, from a UX perspective it makes complete sense IMO to disallow both until we actually support both and have fully tested them, etc.

Reply:
Right now on `feature/autoqasm`, a while loop with a `break` statement fails because we don't yet support `and`

Reply:
More concretely, I can reproduce the same error with these inputs:
```
def main():
    while measure(0):
        aq.gates.x(0)
        break
```
and
```
def main():
    while measure(0) and measure(1):
        aq.gates.x(0)
```

Also, there's some unexpected behavior that `measure(0)` is only executed once, and the same output is referenced every time the condition is evaluated, so I'll open a bug report for that

Reply:
https://github.com/orgs/amazon-braket/projects/2/views/6?pane=issue&itemId=35372617